File: app/src/main/java/com/nobrain/samples/daumeimagesearch/network/retrofit/RetrofitCreator.java
package com.nobrain.samples.daumeimagesearch.network.retrofit;

import retrofit2.Retrofit;
import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitCreator {

    public static Retrofit createRetrofit() {

        return new Retrofit.Builder()
                .baseUrl("https://apis.daum.net/search/")
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
                .build();

    }
}

File: app/src/main/java/com/nobrain/samples/daumeimagesearch/network/retrofit/RetrofitCreator.kt
package com.nobrain.samples.daumeimagesearch.network.retrofit

import retrofit2.Retrofit
import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitCreator {

    fun createRetrofit(): Retrofit {
        return Retrofit.Builder().apply {
            baseUrl("https://apis.daum.net/search/")
            addConverterFactory(GsonConverterFactory.create())
            addCallAdapterFactory(RxJavaCallAdapterFactory.create())
        }.build()


    }
}
